llvm/lib/Target/RISCV/RISCVInstrInfoXTHeadVVLPatterns.td

```diff
@@ -0,0 +1,32 @@
+//===-- RISCVInstrInfoXTHeadVVLPatterns.td - RVV VL patterns -----*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------------===//
+///
+/// This file contains the required infrastructure and VL patterns to support
+/// code generation for the standard 'V' (Vector) extension, version 0.7.1
+///
+/// This file is included from RISCVInstrInfoXTHeadVPseudos.td
+//===---------------------------------------------------------------------------===//
+
+def riscv_th_vmv_v_x_vl : SDNode<"RISCVISD::TH_VMV_V_X_VL",
+                              SDTypeProfile<1, 3, [SDTCisVec<0>, SDTCisInt<0>,
+                                                   SDTCisSameAs<0, 1>,
```

Comment:
I am curious why `vmv.v.v` does not need an ISD node. Might be that vv case is simple enough?

Reply:
We need an ISD node for `vmv.v.x` mainly because of the corner case of loading a 64-bit integer on RV32. 

The 64-bit integer will be split and passed in two 32-bit registers in the generated assembly code. But in MIR, the 64-bit integer will be represented as a vector 64 bits wide with one element. This doesn't matter for other intrinsics, because the intrinsics are overloaded for different type of arguments, and the original `.x` version simply fall through to `.v` version.

But for the vmv intrinsic, we distinguish the `.x` and `.v` version by their name, and llc will complain about the argument type mismatch. To keep the types correct in MIR, one way is to overload `riscv_th_vmv_v_x` to accept vector arguments. But I finally chose to follow what they did in RVV 1.0, because we can then reuse the lowering and optimizing logic of RVV 1.0.

Actually there is an ISD node for `vmv.v.v` but it is used to lower LLVM public ISD nodes such as `INSERT_SUBVECTOR` and `VECTOR_SHUFFLE`. Thus we have nothing to do with its 0.7.1 version for the time being.